src/xinspect.hpp

```diff
@@ -72,13 +72,28 @@ namespace xeus
         std::string typeString;
 
         // add typeinfo in include files in order to use typeid
-        std::string code = "#include<typeinfo>";
+        std::string code = "#include <typeinfo>";
         m_processor.process(code.c_str(), compilation_result, &result);
 
         // try to find the typename of the class
         code = "typeid("+ expression + ").name();";
 
-        if (m_processor.process(code.c_str(), compilation_result, &result))
+        // Temporarily dismissing all std::cerr and std::cout resulting from `m_processor.process`
+        auto errorlevel = 0;
+        {
+            auto cout_strbuf = std::cout.rdbuf();
+            auto cerr_strbuf = std::cerr.rdbuf();
+            auto null = xnull();
+            std::cout.rdbuf(&null);
+            std::cerr.rdbuf(&null);
+
+            errorlevel = m_processor.process(code.c_str(), compilation_result, &result);
+
+            std::cout.rdbuf(cout_strbuf);
+            std::cerr.rdbuf(cerr_strbuf);
+        }
```

Comment:
This is the code that temporarily dismisses the output from `std::cerr` and `std::cout` during the call to `m_processor.process()` required for getting the type id.